### PR TITLE
GH-747: review 3

### PR DESCRIPTION
**Pull request by bertllll**

### CODE REVIEW

File: node/src/accountant/db_access_objects/receivable_dao.rs [hunk 1]
```diff
@@ -37,12 +37,6 @@ pub enum ReceivableDaoError {
     RusqliteError(String),
 }
 
-// impl From<PersistentConfigError> for ReceivableDaoError {
```
Comment (utkarshg6) [inline node/src/accountant/db_access_objects/receivable_dao.rs:40]: Housekeeping :broom: : Just some commented code

File: node/src/accountant/db_access_objects/receivable_dao.rs [hunk 2]
```diff
@@ -1031,16 +1023,19 @@ mod tests {
...
+            PrepareMethodResultsProducer::construct_with_prod_code_and_altered_stmts(
+                prod_code_calls_conn,
+                Some(panic_causing_conn),
+                vec![
+                    StmtTypeDirective::ExecuteProdCode,
+                    StmtTypeDirective::ExecuteProdCode,
+                    StmtTypeDirective::ExecuteProdCode,
+                    StmtTypeDirective::UseAlteredStmt(
+                        AlteredStmtBySQLOrigin::SQLIdenticalWithProdCode,
+                    ),
```
Comment (utkarshg6) [inline node/src/accountant/db_access_objects/receivable_dao.rs:1036]: Nicely done :tada:

File: node/src/accountant/db_access_objects/receivable_dao.rs [hunk 3]
```diff
@@ -1168,52 +1163,6 @@ mod tests {
         assert_eq!(result, Err(RusqliteError(expected_panic_msg.to_string())))
     }
 
-    #[test]
-    fn verify_possibly_unknown_wallet_returns_error_finding_duplicated_records_for_single_wallet() {
```
Comment (utkarshg6) [inline node/src/accountant/db_access_objects/receivable_dao.rs:1172]: I don't know, why this test was removed.

**Review by utkarshg6 — COMMENTED** [review]
Continue at node/src/accountant/db_big_integer/big_int_db_processor.rs:525

**Review by utkarshg6 — COMMENTED** [review]
Continue at node/src/accountant/db_big_integer/big_int_db_processor.rs:915

File: node/src/database/rusqlite_wrappers.rs [hunk 4]
```diff
@@ -1,58 +1,52 @@
...
-// did write some code using it, we always acknowledged we had to leave that piece of logic untested.
-// The actual problem with a mocked transaction has been that we usually need to combine it with
-// an also mocked connection. The transaction is springs from the connection. The need to put
-// this object, strictly depending on a reference, into another mock makes us face a nontrivial
-// relationship of lifetimes, growing in their count as a chase after what the compiler asks of is
-// unwinding.
+// The truth is the use of an explicit, object-like, transaction from 'rusqlite' is rare for our code.
+// That might be why we'd managed to live so long without being much constrained by absence of
+// the mock. While we did write some code using that kind of a transaction, we always acknowledged
+// only that we had to make and exception and leave that piece of code untested. The problem
```
Comment (utkarshg6) [inline node/src/database/rusqlite_wrappers.rs:26]: **an** instead of **and**

File: node/src/database/rusqlite_wrappers.rs [hunk 5]
```diff
@@ -1,58 +1,52 @@
...
+// One may think about creating another class to wrap up the `Statement` to have a better control.
+// Tested, but its interface requires even more methods and so obstacles, whereupon the steeply
+// increasing difficulty ended the efforts.
 //
-// It is fair to admit our mock is poor at elegance and simplicity. Especially when it takes more
-// complex tests. However, compensation for that needs to be seen in the good portion of
-// possibilities it opens up for advance, thorough testing in those rare places where we do rely on
-// an explicit 'rusqlite' transaction and also let's keep in mind it used to seem undoable not so
-// far ago.
+// Fair to admit the mock is poor at elegance and simplicity. The more the more complex tests it
```
Comment (utkarshg6) [inline node/src/database/rusqlite_wrappers.rs:44]: Let's rephrase it to say, "Fair to admit that this mock is a gift to our codebase and it opens up the possibilities for thorough testing in those rare places..." (you may continue your paragraph from here)

:wink:

**Review by utkarshg6 — COMMENTED** [review]
Continue at node/src/database/rusqlite_wrappers.rs:101

File: node/src/database/test_utils/transaction_wrapper_mock.rs [hunk 6]
```diff
@@ -12,21 +12,20 @@ use rusqlite::{Error, Statement, ToSql};
...
-// creates an encapsulated environment in which it configures and eventually builds the mock. All
-// that in order to minimize the exposure of the mock
-// to the outer scope. While a wrapped transaction is being created, and we need it to be tamed in
-// a test, this mock is supposed to become its inner wrapper of the supposed real world transaction.
-// The inner wrapper ever needs to stay private. Then there is the outer, public one, that anyone
-// can manipulate with.
+// The qualities of this builder come from the freedom of its usability contrasting with how it
+// creates an encapsulated environment inside which it configures and eventually builds
+// the requested mock. This happens in order to minimize the exposure of the special mock to
+// the outer scope. To such an extend that any developer should become ever that things have been
```
Comment (utkarshg6) [inline node/src/database/test_utils/transaction_wrapper_mock.rs:18]: should become **aware**

File: node/src/database/test_utils/transaction_wrapper_mock.rs [hunk 7]
```diff
@@ -140,42 +120,37 @@ impl TransactionInnerWrapper for TransactionInnerWrapperMock {
...
 // Our focus must be on the Statement produced by this method. It needs to be understood that
-// the 'prepare' method has a crucial influence to the result of the following function call taking
-// the Statement as an argument. Fortunately, despite some challenges being always around, we can
-// indirectly steer the course of that future procedure to have the very result we want to happen
-// in the test. This approach can be applied to various methods such as 'execute', 'query_row',
-// or 'query_map'.
+// the 'prepare' method has a crucial influence on the result of the following function, usually
+// executing this prepared 'Statement'. Fortunately, still with some challenges being always around,
+// we can indirectly steer the course of events of that future call and rely on an exact result to
+// to happen in the test. This approach can be applied for all respective methods such as 'execute',
```
Comment (utkarshg6) [inline node/src/database/test_utils/transaction_wrapper_mock.rs:147]: The `to` is repeated

**Review by utkarshg6 — COMMENTED** [review]
Continue at node/src/database/test_utils/transaction_wrapper_mock.rs:166

File: node/src/database/test_utils/transaction_wrapper_mock.rs [hunk 8]
```diff
@@ -313,31 +291,30 @@ impl PrepareResultsDispatcher {
...
-                new_stmt: incident_statement,
-            } => incident_statement.as_str(),
-        }
-    }
+#[derive(Debug)]
+pub enum AlteredStmtBySQLOrigin {
+    // Use this on planning to have a conn pointing to a different db. This will eliminate having to
+    // think up a new stmt, instead, the same one as in the prod code will be used.
+    SQLIdenticalWithProdCode,
+    Substitution { new_stmt: String },
```
Comment (utkarshg6) [inline node/src/database/test_utils/transaction_wrapper_mock.rs:319]: Well Done :clap:

**Review by utkarshg6 — APPROVED** [review]

File: node/src/accountant/db_access_objects/receivable_dao.rs (hunk 3 above)
Comment (utkarshg6) [reply node/src/accountant/db_access_objects/receivable_dao.rs:1172]: I found this test is failing - `verify_possibly_unknown_wallet_returns_error_when_the_row_for_this_wallet_exists`.

So, I believe it was alright to remove the test.

File: node/src/accountant/db_access_objects/receivable_dao.rs (hunk 1 above)
Comment (bertllll) [reply node/src/accountant/db_access_objects/receivable_dao.rs:40]: It was already removed in the previous iteration ;-)

File: node/src/database/rusqlite_wrappers.rs (hunk 5 above)
Comment (bertllll) [reply node/src/database/rusqlite_wrappers.rs:44]: I was thinking that you probably had been just joking, but the I also corrected this paragraph, hopefully you'll be okay with it.
